src/components/Trade/utils.ts

```diff
@@ -0,0 +1,19 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import { getInboundAddressDataForChain, SwapperName } from '@shapeshiftoss/swapper'
+import { getConfig } from 'config'
+
+export const isTradingActive = async (
+  assetId: AssetId | undefined,
+  swapperName: SwapperName,
+): Promise<boolean> => {
+  switch (swapperName) {
+    case SwapperName.Thorchain: {
+      const daemonUrl = getConfig().REACT_APP_THORCHAIN_NODE_URL
+      const inboundAddressData = await getInboundAddressDataForChain(daemonUrl, assetId, false)
+      // If we can't get inbound address data, we MUST assume trading is halted for safety
+      return inboundAddressData ? !inboundAddressData.halted : false
```

Comment:
This would work as a shorter one-liner, though I do prefer the explicitness of that left side if that was the intent!
```suggestion
      return !(inboundAddressData.halted)
```